packages/amplify_analytics_pinpoint/ios/Classes/AmplifyAnalyticsBuilder.swift

```diff
@@ -18,8 +18,27 @@ import AmplifyPlugins
 
 public class AmplifyAnalyticsBuilder {
 
-    public static func createAnalyticsProperties(propertiesMap : Dictionary<String, Any>) -> AnalyticsProperties{
-        return propertiesMap as? AnalyticsProperties ?? [:]
+    public static func createAnalyticsProperties(map : [String: Any]) -> AnalyticsProperties{
+        let propertiesMap = map["propertiesMap"] as! [String: Any]
+        let propertiesTypesMap = map["propertiesTypesMap"] as! [String: String]
+        
+        var analyticsProperties = AnalyticsProperties()
+        
+        for (key,value) in propertiesTypesMap {
+            switch value{
+                case "STRING":
+                    analyticsProperties[key] = propertiesMap[key] as! String
+                case "INT":
+                    analyticsProperties[key] = propertiesMap[key] as! Int
+                case "DOUBLE":
+                    analyticsProperties[key] = propertiesMap[key] as! Double
+                case "BOOL":
+                    analyticsProperties[key] = propertiesMap[key] as! Bool
+                default:
+                    print("Unknown type for AnalyticsProperties")
+            }
```

Comment:
If we can avoid force unwrap, we should do that

Reply:
propertiesTypesMap needs to be force unwrapped as it's used for the foreach.  But I can use ? for String/Int/Double/Bool casts.